### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by yanaair**
- added sorting of messages before logging errors
- changed error messages to a more readable format, added error count to logs
- added parameter for build not to fail if there are validation errors (log them as warnings instead)
- improved visiting of model properties
- some minor fixes that were not noticed before

### CODE REVIEW

File: src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/ValidateMojo.java [hunk 1]
```diff
@@ -38,6 +39,9 @@
     @Parameter
     private String customPathValidatorsPackage;
 
+    @Parameter(defaultValue = "true")
+    private Boolean failOnErrors;
```
Comment (sylvainlaurent) [inline src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/ValidateMojo.java]: Boolean -> boolean

File: README.md [hunk 2]
```diff
@@ -45,7 +46,7 @@ This maven plugin allows to validate yaml and json files.
       </plugin>
 ```
 
-Validation failures make the build fail.
+Validation failures make the build fail but default. You can change this by setting `<failOnErrors>false</failOnErrors>` and error messages will be logged as warnings.
```
Comment (sylvainlaurent) [inline README.md]: At first sight when I see such a param as failOnErrors, I just expect errors to remain errors but just avoid failing te build. So, the severity should not change from error to warning.

Comment (yanaair) [reply README.md]: You are right, fixed

File: src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/ValidateMojo.java (hunk 1 above)
Comment (yanaair) [reply src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/ValidateMojo.java]: Changed

**Comment by yanaair** [comment]
@sylvainlaurent hello, I have implemented your review comments :)

**Pull request merged**